clients/src/test/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinatorTest.java

```diff
@@ -259,6 +260,22 @@ public void testCoordinatorDiscoveryBackoff() {
         assertTrue(endTime - initialTime >= RETRY_BACKOFF_MS);
     }
 
+    @Test
+    public void testGroupAuthorizationFailure() {
+        setupCoordinator();
+
+        mockClient.prepareResponse(groupCoordinatorResponse(node, Errors.GROUP_AUTHORIZATION_FAILED));
+
+        long initialTimeMs = mockTime.milliseconds();
+        assertThrows(GroupAuthorizationException.class, () ->
+                coordinator.ensureCoordinatorReady(mockTime.timer(Long.MAX_VALUE)));
+
+        mockClient.prepareResponse(groupCoordinatorResponse(node, Errors.NONE));
+        coordinator.ensureCoordinatorReady(mockTime.timer(Long.MAX_VALUE));
+        long timeTakenMs = mockTime.milliseconds() - initialTimeMs;
```

Comment:
Should we do the timeTaken check immediately after L270 assertion to indicate that we are testing on the back-off triggered for group authorization failure? The current flow looks a bit weird as the second `ensureCoordinatorReady` doesn't do any back-off.

Reply:
Done.